rclpy/rclpy/__init__.py

```diff
@@ -231,15 +233,17 @@ def spin(node: 'Node', executor: Optional['Executor'] = None) -> None:
     This function blocks.
 
     :param node: A node to add to the executor to check for work.
-    :param executor: The executor to use, or the global executor if ``None``.
+    :param executor: The executor to use, or the node's current or the global executor if ``None``.
     """
-    executor = get_global_executor() if executor is None else executor
+    executor = executor or node.executor or get_global_executor()
```

Comment:
although this makes https://github.com/ros2/rclpy/issues/1018#issue-1395363459 works with `node.executor`,

If `executor` is specified, and `node.executor` is assigned, this will replace the `node.executor` implicitly. i do not think this is expected behavior by user. instead, we should generate the exception to let the user know that this node has been already associated with executor? i agree with https://github.com/ros2/rclpy/issues/1018#issuecomment-1280524111.